app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java

```diff
@@ -181,55 +190,205 @@ private void requestNotificationPermission() {
     @Override
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
-        if (intent.getExtras() != null){
-            if (intent.getExtras().containsKey(RES_ITEM))
-                binding.bottomNavigation.setSelectedItemId(intent.getIntExtra(RES_ITEM, R.id.dashboard));
-            else if (intent.getExtras().containsKey(NOTIFICATION_DIALOG)){
-                new ConfirmDialogFragment.Builder()
-                        .withTitle(intent.getExtras().getString("title"))
-                        .withMessage(intent.getExtras().getString("message"))
-                        .withNegativeButton("")
-                        .withPositiveButton(getString(R.string.Modal_OK))
-                        .build().show(getSupportFragmentManager(), null);
+        /**
+         * Check if we are starting the activity from a link [Intent.ACTION_VIEW].
+         * This is invoked when a v2 link is opened.
+         */
+        if (Intent.ACTION_VIEW.equals(intent.getAction())) {
+            Uri uri = intent.getData();
+            // If the intent does not contain a link, do nothing.
+            if (uri == null) {
+                return;
+            }
+
+            long runId = getRunId(uri);
+
+            // If the intent contains a link, but the `link_id` is zero, or the link is not supported, do nothing.
+            if (runId == 0) {
```

Comment:
If you want to stick with returning an integer pattern, this would probably be:

```suggestion
            if (runId <= 0) {
```

To exclude all the numbers that don't belong to the domain